apps/showcase/src/components/showcase/rules-engine/rules-engine-pres.component.ts

```diff
@@ -46,50 +50,43 @@ export class RulesEnginePresComponent implements OnDestroy, OnChanges, DynamicCo
-
-  /** Observable of the destination */
-  public destinations$: Observable<RulesEngineDestinationConfiguration[]>;
-
-  constructor(
-    @Optional() configurationService: ConfigurationBaseService,
-      fb: FormBuilder,
-      tripService: TripFactsService,
-      localizationService: LocalizationService
-  ) {
-    this.dynamicConfig$ = new ConfigurationObserver<RulesEnginePresConfig>(RULES_ENGINE_PRES_CONFIG_ID, RULES_ENGINE_PRES_DEFAULT_CONFIG, configurationService);
-    this.config$ = this.dynamicConfig$.asObservable();
-    this.destinations$ = this.config$.pipe(map((config) => config.destinations));
-    this.form = fb.group({
+  }> = inject(FormBuilder).group({
       destination: new FormControl<string | null>(null),
       outboundDate: new FormControl<string | null>(this.formatDate(Date.now() + 7 * ONE_DAY_IN_MS)),
       inboundDate: new FormControl<string | null>(this.formatDate(Date.now() + 14 * ONE_DAY_IN_MS))
     });
-    this.subscription.add(this.form.controls.destination.valueChanges.subscribe((destination) => tripService.updateDestination(destination)));
-    this.subscription.add(this.form.controls.outboundDate.valueChanges.subscribe((outboundDate) => tripService.updateOutboundDate(outboundDate)));
-    this.subscription.add(this.config$.pipe(map(({ inXDays }) => inXDays), distinctUntilChanged()).subscribe((inXDays) => {
-      this.form.controls.outboundDate.setValue(this.formatDate(Date.now() + inXDays * ONE_DAY_IN_MS));
-      if (this.form.value.inboundDate && this.form.value.outboundDate && this.form.value.inboundDate <= this.form.value.outboundDate) {
-        this.form.controls.inboundDate.setValue(this.formatDate((this.form.value.outboundDate ? (new Date(this.form.value.outboundDate)).getTime() : Date.now()) + 7 * ONE_DAY_IN_MS));
-      }
-    }));
-    this.subscription.add(this.config$.pipe(map(({ destinations }) => destinations)).subscribe((destinations) => {
-      const selectedDestination = destinations.find((d) => d.cityName === this.form.value.destination);
-      if (selectedDestination && !selectedDestination.available) {
-        this.form.controls.destination.reset();
-      }
-    }));
-    this.translations = translations;
+
+  /** Input configuration to override the default configuration of the component */
+  public config = input<Partial<RulesEnginePresConfig>>();
+
+  @O3rConfig()
+  public readonly configSignal = configSignal(
+      this.config,
+      RULES_ENGINE_PRES_CONFIG_ID,
+      RULES_ENGINE_PRES_DEFAULT_CONFIG
+    );
+
+  constructor() {
+    this.subscription.add(this.form.controls.destination.valueChanges.subscribe((destination) => this.tripService.updateDestination(destination)));
+    this.subscription.add(this.form.controls.outboundDate.valueChanges.subscribe((outboundDate) => this.tripService.updateOutboundDate(outboundDate)));
+    const inXDays$ = toObservable(computed(() => this.configSignal().inXDays));
+    this.subscription.add(
+      inXDays$.subscribe((inXDays) => {
```

Comment:
Why don't we use an effect here?

Reply:
This is also true for the `destination$` object

Reply:
It can be done in another PR to keep it simple as it's not linked to the usage of `inject`